### PR TITLE
Apply PreferAssertJ suggested fixes

### PR DESCRIPTION
**Pull request by carterkozak**
Applied suggested fixes, then made a few manual fixes where generics went wonky.

### CODE REVIEW

File: atlasdb-api/src/test/java/com/palantir/atlasdb/keyvalue/api/SweepResultsTest.java [hunk 1]
```diff
@@ -64,52 +66,49 @@
 
     @Test
     public void accumulateWithNextRowInOrder() {
-        Assert.assertThat(RESULTS.accumulateWith(OTHER_RESULTS), Matchers.equalTo(COMBINED_RESULTS_AND_OTHER));
+        assertThat(RESULTS.accumulateWith(OTHER_RESULTS)).is(new HamcrestCondition<>(Matchers.equalTo(COMBINED_RESULTS_AND_OTHER)));
```
Comment (schlosna) [inline atlasdb-api/src/test/java/com/palantir/atlasdb/keyvalue/api/SweepResultsTest.java]: Should we update the refactor tolling to transform `Matchers.equalTo` to `isEqualTo`?

Comment (carterkozak) [reply atlasdb-api/src/test/java/com/palantir/atlasdb/keyvalue/api/SweepResultsTest.java]: Yep, that's reasonable. I don't think we'll be able to support migration of all hamcrest matchers, but the most common ones should be doable.

Comment (carterkozak) [reply atlasdb-api/src/test/java/com/palantir/atlasdb/keyvalue/api/SweepResultsTest.java]: Added support for migrating common hamcrest matchers here: https://github.com/palantir/gradle-baseline/pull/850

**Comment by jeremyk-91** [comment]
@carterkozak Would it be plausible to separate the uncontentious changes from the ones along the lines of the Hamcrest matchers which are arguably negative? Thanks!

**Pull request closed without merging**